packages/@aws-cdk/pipelines/lib/codepipeline/codepipeline.ts

```diff
@@ -0,0 +1,64 @@
+import * as cb from '@aws-cdk/aws-codebuild';
+import * as cp from '@aws-cdk/aws-codepipeline';
+import { Construct } from 'constructs';
+import { Step } from '../blueprint';
+import { Pipeline } from '../main/pipeline';
+import { CodePipelineEngine, CodePipelineEngineProps } from './codepipeline-engine';
+
+
+/**
+ * Properties for a `CodePipelinePipeline`
+ */
+export interface CodePipelinePipelineProps extends CodePipelineEngineProps {
+  /**
+   * The build step that produces the CDK Cloud Assembly
+   *
+   * The primary output of this step needs to be the `cdk.out` directory
+   * generated by the `cdk synth` command.
+   *
+   * If you use `SynthStep`, the output directory will automatically be
+   * configured correctly; if you use a different kind of `Step`, you
+   * may need to configure it explicitly.
+   */
+  readonly synthStep: Step;
+}
+
+/**
+ * A CDK Pipeline that uses CodePipeline to deploy CDK apps
+ *
+ * This is a `Pipeline` with its `engine` property set to
+ * `CodePipelineEngine`, and exists for nicer ergonomics for
+ * users that don't need to switch out engines.
+ */
+export class CodePipelinePipeline extends Pipeline {
```

Comment:
I understand the purity, but `CodePipeline` should be enough, no?


```suggestion
export class CodePipeline extends Pipeline {
```

Reply:
I had that first, was wondering if it wasn't too confusing.